### PR TITLE
[velero] Add doc on how to include secrets in values.yaml

### LINKED ISSUE #28: Add documentation to values.yaml on how to include secrets
I am installing Velero in K8s cluster that will use Azure Storabe blob as backed, using latest chart version 2.7.5, but I am getting the following error in Velero logs on startup:

```
time="2020-01-02T11:33:46Z" level=info msg="setting log-level to INFO" logSource="pkg/cmd/server/server.go:171"
time="2020-01-02T11:33:46Z" level=info msg="Starting Velero server v1.2.0 (5d008491bbf681658d3e372da1a9d3a21ca4c03c)" logSource="pkg/cmd/server/server.go:173"
time="2020-01-02T11:33:46Z" level=info msg="No feature flags enabled" logSource="pkg/cmd/server/server.go:177"
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pod
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pv
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=BackupItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-account
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pv-from-pvc
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/add-pvc-from-pod
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/change-storage-class
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/cluster-role-bindings
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/job
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/pod
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/restic
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/role-bindings
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/velero kind=RestoreItemAction logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/service-account
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/plugins/velero-plugin-for-microsoft-azure kind=VolumeSnapshotter logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/azure
time="2020-01-02T11:33:46Z" level=info msg="registering plugin" command=/plugins/velero-plugin-for-microsoft-azure kind=ObjectStore logSource="pkg/plugin/clientmgmt/registry.go:100" name=velero.io/azure
time="2020-01-02T11:33:46Z" level=info msg="Checking existence of namespace" logSource="pkg/cmd/server/server.go:337" namespace=velero
time="2020-01-02T11:33:46Z" level=info msg="Namespace exists" logSource="pkg/cmd/server/server.go:343" namespace=velero
time="2020-01-02T11:33:49Z" level=info msg="Checking existence of Velero custom resource definitions" logSource="pkg/cmd/server/server.go:372"
time="2020-01-02T11:33:49Z" level=info msg="All Velero custom resource definitions exist" logSource="pkg/cmd/server/server.go:406"
time="2020-01-02T11:33:49Z" level=info msg="Checking that all backup storage locations are valid" logSource="pkg/cmd/server/server.go:413"
An error occurred: some backup storage locations are invalid: error getting backup store for location "default": rpc error: code = Unknown desc = error loading environment from AZURE_CREDENTIALS_FILE (/credentials/cloud): open /credentials/cloud: no such file or directory
```

Here is my valuesfile:

```
##
## Configuration settings that directly affect the Velero deployment YAML.
##

# Details of the container image to use in the Velero deployment & daemonset (if
# enabling restic). Required.
image:
  repository: velero/velero
  tag: v1.2.0
  pullPolicy: IfNotPresent

# Annotations to add to the Velero deployment's pod template. Optional.
#
# If using kube2iam or kiam, use the following annotation with your AWS_ACCOUNT_ID
# and VELERO_ROLE_NAME filled in:
#  iam.amazonaws.com/role: arn:aws:iam::<AWS_ACCOUNT_ID>:role/<VELERO_ROLE_NAME>
podAnnotations: {}

# Resource requests/limits to specify for the Velero deployment. Optional.
resources: {}

# Init containers to add to the Velero deployment's pod spec. At least one plugin provider image is required.
initContainers:
   - name: velero-plugin-for-microsoft-azure
     image: velero/velero-plugin-for-microsoft-azure:v1.0.0
     imagePullPolicy: IfNotPresent
     volumeMounts:
       - name: plugins
         mountPath: /target

# Tolerations to use for the Velero deployment. Optional.
tolerations: []

# Node selector to use for the Velero deployment. Optional.
nodeSelector: {}

# Extra volumes for the Velero deployment. Optional.
extraVolumes:

# Extra volumeMounts for the Velero deployment. Optional.
extraVolumeMounts: []

# Settings for Velero's prometheus metrics. Enabled by default.
metrics:
  enabled: true
  scrapeInterval: 15s

  # Pod annotations for Prometheus
  podAnnotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "8085"
    prometheus.io/path: "/metrics"

  serviceMonitor:
    enabled: false
    additionalLabels: {}

##
## End of deployment-related settings.
##



##
## Parameters for the `default` BackupStorageLocation and VolumeSnapshotLocation,
## and additional server settings.
##
configuration:
  provider: azure

  # Parameters for the `default` BackupStorageLocation. See
  # https://velero.io/docs/v1.0.0/api-types/backupstoragelocation/
  backupStorageLocation:
    name: azure
    bucket: <redacted>
    prefix: test
    config:
      resourceGroup: <redacted>
      storageAccount: <redacted>

  # These are server-level settings passed as CLI flags to the `velero server` command. Velero
  # uses default values if they're not passed in, so they only need to be explicitly specified
  # here if using a non-default value. The `velero server` default values are shown in the
  # comments below.
  # --------------------
  # `velero server` default: 1m
  backupSyncPeriod: 60m
  # `velero server` default: 1h
  resticTimeout:
  # `velero server` default: namespaces,persistentvolumes,persistentvolumeclaims,secrets,configmaps,serviceaccounts,limitranges,pods
  restoreResourcePriorities:
  # `velero server` default: false
  restoreOnlyMode:

  # additional key/value pairs to be used as environment variables such as "AWS_CLUSTER_NAME: 'yourcluster.domain.tld'"
  extraEnvVars: {}

  # Set log-level for Velero pod. Default: info. Other options: debug, warning, error, fatal, panic.
  logLevel:

  # Set log-format for Velero pod. Default: text. Other option: json.
  logFormat:

##
## End of backup/snapshot location settings.
##



##
## Settings for additional Velero resources.
##

# Whether to create the Velero cluster role binding.
rbac:
  create: true

# Information about the Kubernetes service account Velero uses.
serviceAccount:
  server:
    create: true
    name:

# Info about the secret to be used by the Velero deployment, which
# should contain credentials for the cloud provider IAM account you've
# set up for Velero.
credentials:
  existingSecret: "cloud-credentials"
  useSecret: true

# Wheter to create volumesnapshotlocation crd, if false => disable snapshot feature
snapshotsEnabled: false

# Whether to deploy the restic daemonset.
deployRestic: false

schedules: 
  daily-backup:
    schedule: "0 17 * * *"
    template:
      ttl: "120h"
      excludedNamespaces:
        - global-infrastructure

configMaps: {}

##
## End of additional Velero resource settings.
##
```

And here are the contents of pre-existing secret "cloud-credentials" in the namespace where velero is being installed:
```
apiVersion: v1
data:
  AZURE_CLIENT_ID: <redacted>
  AZURE_CLIENT_SECRET: <redacted>
  AZURE_CLOUD_NAME: <redacted>
  AZURE_RESOURCE_GROUP: <redacted>
  AZURE_SUBSCRIPTION_ID: <redacted>
  AZURE_TENANT_ID: <redacted>
kind: Secret
metadata:
  creationTimestamp: "2020-01-02T11:07:22Z"
  labels:
    CREATED_BY: terraform
  name: cloud-credentials
  namespace: velero
  resourceVersion: "43307401"
  selfLink: /api/v1/namespaces/velero/secrets/cloud-credentials
  uid: 0d031617-2d50-11ea-a618-e29387fbefef
type: Opaque
```

Am I doing anything wrong with my configuration?

### PR DESCRIPTION
**Pull request by jenting**
#### Special notes for your reviewer:

fixes #28

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[velero]`)

### CODE REVIEW

File: charts/velero/values.yaml [hunk 1]
```diff
@@ -204,6 +204,30 @@ credentials:
...
+  #   AZURE_TENANT_ID=<REDACTED>
+  #   AZURE_CLIENT_ID=<REDACTED>
+  #   AZURE_CLIENT_SECRET=<REDACTED>
+  #   AZURE_RESOURCE_GROUP=<REDACTED>
+  #   AZURE_CLOUD_NAME=AzurePublicCloud
+  # cloud: |
+  #   {
+  #     "type": "<REDACTED>",
+  #     "project_id": "<REDACTED>",
+  #     "private_key_id": "<REDACTED>",
```
Comment (pantelis-karamolegkos) [inline charts/velero/values.yaml:222]: Would it be possible to have a comment / note about what is optional vs mandatory? For example are *both* the `private_key_id` and `private_key` needed?

Comment (jenting) [reply charts/velero/values.yaml:222]: NP!

**Comment by pantelis-karamolegkos** [comment]
Hello @jenting , I have tried the above format but I am having this [issue](https://github.com/vmware-tanzu/helm-charts/issues/180), would it be possible to get some help if/whenever possible? Many thanks in advance.

File: charts/velero/values.yaml (hunk 1 above)
Comment (jenting) [reply charts/velero/values.yaml:222]: For GCP, the _credentials-velero_ is fetched from gcloud CLI, ref to https://github.com/vmware-tanzu/velero-plugin-for-gcp#set-permissions-for-velero. So, I think all are mandatory.

However, I add the available values for `AZURE_CLOUD_NAME`.

**Comment by jenting** [comment]
@carlisia @cpanato PTAL

**Review by cpanato — DISMISSED** [review]

**Review by cpanato — APPROVED** [review]

**Review by carlisia — CHANGES_REQUESTED** [review]
This example definitely needed (I actually thought we had already added it to the values.yaml file).

However, I would like it for us to avoid replicating the provider documentation. Any specific value of information if it's optional or required should be looked up at the provider's docs.

The main thing that trips people up is the key of the secret needing to be named `cloud`.

**Review by carlisia — CHANGES_REQUESTED** [review]
I think this should be sufficient:


```
  # secretContents contains the data to be stored in the Velero secret, if `useSecret` is
  # true and `existingSecret` is empty. Please note: as of the current Velero release,
  # Velero only uses one secret key/value at a time, and the key must be named
  # `cloud`. The value corresponds to the entire content of your IAM credentials file. See the example
  # below. Note that the format will be different for different providers, please check
  # their documentation.
  secretContents: {}	  secretContents: {}
  # cloud: |
  #   [default]
  #   aws_access_key_id=<REDACTED>
  #   aws_secret_access_key=<REDACTED>
```

**Comment by jenting** [comment]
I think it'd be better to add the velero supported provider plugin README in the comment, WDYT?
```
# secretContents contains the data to be stored in the Velero secret, if `useSecret` is
# true and `existingSecret` is empty.
# Please note: as of the current Velero release, Velero only uses one secret key/value at a time,
# and the key must be named `cloud`. The value corresponds to the entire content of your IAM credentials file.
# See the example below. Note that the format will be different for different providers, please check
# their documentation.
# [AWS] https://github.com/vmware-tanzu/velero-plugin-for-aws/blob/main/README.md
# [GCP] https://github.com/vmware-tanzu/velero-plugin-for-gcp/blob/main/README.md
# [Azure] https://github.com/vmware-tanzu/velero-plugin-for-microsoft-azure/blob/main/README.md
secretContents: {}
#  cloud: |
#    [default]
#    aws_access_key_id=<REDACTED>
#    aws_secret_access_key=<REDACTED>
```

**Comment by carlisia** [comment]
That works for me. But since we have more plugins then that, to explicit try and avoid confusion, please change this line:

" Note that the format will be different for different providers, please check their documentation."

To:

" Note that the format will be different for different providers, please check their documentation. Here is a list of documentation for plugins maintained by the Velero team:"

**Review by carlisia — APPROVED** [review]
Thank you 👍

**Review by carlisia — CHANGES_REQUESTED** [review]
Oops, a trilling space issue: https://github.com/vmware-tanzu/helm-charts/pull/177/checks?check_run_id=1567640462#step:4:111.

**Review by carlisia — APPROVED** [review]
Thank you for this! 👍

**Comment by carlisia** [comment]
@cpanato / @nrb this is a no brainer, would love to merge it.

**Review by cpanato — APPROVED** [review]

**Pull request merged**